lint-configs/symlink-cpp-lint-configs.sh

```diff
@@ -15,7 +15,12 @@ symlink_config () {
     config_file_path="$1"
 
     repo_dir="$(git rev-parse --show-toplevel)"
-    repo_relative_config_file_path="$(realpath --relative-to="$repo_dir" "$config_file_path")"
+    config_file_absolute_path="$(readlink -f "$config_file_path")"
+
+    # NOTE: This is a bit fragile since it depends on adding a trailing slash to $repo_dir. Ideally,
+    # we would use `realpath --relative-to` instead of variable substitution, but `--relative-to`
+    # isn't available on macOS.
+    repo_relative_config_file_path="${config_file_absolute_path#"${repo_dir}/"}"
 
     src_path="${repo_dir}/${repo_relative_config_file_path}"
```

Comment:
Isn't this always the same as `config_file_path`?

Reply:
Not necessarily since `config_file_path` might contain a symlink along the path. E.g.:

`a/b/c/.clang-format` where `c` is actually a symlink to `a/d`. So `config_file_path` would be `a/b/c/.clang-format` but `config_file_absolute_path` would be `a/d/.clang-format` and `src_path` would be `${repo_dir}/a/d/.clang-format`.